src/languages/smtlib/parseSmtlib.mly

```diff
@@ -137,60 +170,109 @@ command_option:
 ;
 
 command:
-  | OPEN POP n=NUMERAL CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.pop ~loc (int_of_string n) }
-  | OPEN PUSH n=NUMERAL CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.push ~loc (int_of_string n) }
-
   | OPEN ASSERT t=term CLOSE
     { let loc = L.mk_pos $startpos $endpos in S.assert_ ~loc t }
   | OPEN CHECK_SAT CLOSE
     { let loc = L.mk_pos $startpos $endpos in S.check_sat ~loc () }
 
-  | OPEN SET_LOGIC s=SYMBOL CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.set_logic ~loc s }
+  | OPEN DECLARE_CONST s=SYMBOL ty=sort CLOSE
+    { let id = I.(mk term s) in
+      let loc = L.mk_pos $startpos $endpos in
+      S.fun_decl ~loc id [] ty }
 
-  | OPEN GET_INFO i=KEYWORD CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.get_info ~loc i }
-  | OPEN SET_INFO c=command_option CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.set_info ~loc c }
+  | OPEN DECLARE_DATATYPE s=SYMBOL OPEN clist=datatype_dec CLOSE CLOSE
+    { let id = I.(mk sort s) in
+      let loc = L.mk_pos $startpos $endpos in
+      S.dtype_decl ~loc id [] (snd clist) }
 
-  | OPEN GET_OPTION k=KEYWORD CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.get_option ~loc k }
-  | OPEN SET_OPTION c=command_option CLOSE
-    { let loc = L.mk_pos $startpos $endpos in S.set_option ~loc c }
+  | OPEN DECLARE_DATATYPES OPEN sl=sort_dec+ CLOSE OPEN lcl=datatype_dec+ CLOSE CLOSE
+    { let loc = L.mk_pos $startpos $endpos in
+      let dl = List.map2 (fun sd cl -> S.dtype_decl ~loc (fst sd) (fst cl) (snd cl)) sl lcl 
```

Comment:
This function is quite hard to read and understand.
It would be better to plit it over multiple lines, and use let-bindings instead of `fst` or `snd`, in order to be more explicit (i.e. give a legible name to the intermediary variables), and be a tinly little bit more efficient (let's call it a personnal preference, ^^)